### PR TITLE
fix build for bug 6429 and snapshot incremental update logic

### PR DESCRIPTION
**Pull request by hcyang**

### CODE REVIEW

File: packages/orchestratorlib/src/predict.ts [hunk 1]
```diff
@@ -1289,7 +1289,7 @@ export class OrchestratorPredict {
     const example: Example = Example.newIntentExample(
       this.currentUtterance,
       this.currentIntentLabels);
-    const exampleObejct: any = example.toObject();
+    const exampleObejct: any = example.toAlternateObject();
```
Comment (tsuwandy) [inline packages/orchestratorlib/src/predict.ts:1292]: exampleObejct-> exampleObject 😊

File: packages/orchestratorlib/src/predict.ts [hunk 2]
```diff
@@ -1309,7 +1309,7 @@ export class OrchestratorPredict {
     const exampleToRemove: Example = Example.newIntentExample(
       this.currentUtterance,
       this.currentIntentLabels);
-    const exampleObejctToRemove: any = exampleToRemove.toObject();
+    const exampleObejctToRemove: any = exampleToRemove.toAlternateObject();
```
Comment (tsuwandy) [inline packages/orchestratorlib/src/predict.ts:1312]: exampleObjectToRemove

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 3]
```diff
@@ -811,6 +812,60 @@ export class OrchestratorHelper {
     return false;
   }
 
+  // eslint-disable-next-line max-params
+  static getExampleArrayIntentsEntitiesUtterances(
+    exampleArray: any,
```
Comment (tsuwandy) [inline packages/orchestratorlib/src/orchestratorhelper.ts:817]: Should this be 
exampleArray: any[]

**Review by tsuwandy — APPROVED** [review]

File: packages/orchestratorlib/src/predict.ts (hunk 1 above)
Comment (hcyang) [reply packages/orchestratorlib/src/predict.ts]: Fixed!

File: packages/orchestratorlib/src/predict.ts (hunk 2 above)
Comment (hcyang) [reply packages/orchestratorlib/src/predict.ts]: Typo fixed!

File: packages/orchestratorlib/src/orchestratorhelper.ts (hunk 3 above)
Comment (hcyang) [reply packages/orchestratorlib/src/orchestratorhelper.ts:817]: It was returned by the getExamples() ABI as any, even though we know it's an example array.

**Pull request merged**